### DISCUSSION TITLE
Incorrect shipping cost calculation with cart rule free gift and percent promocode

### DISCUSSION THREAD

**Discussion by AnnRise** [PrestaShop/PrestaShop#34408] [Q&A]
To reproduce the situation:
1) Create product 1 with price 30 (tax included 20%)
2) Create product 2 with price 100 (tax included 20%)
3) Create carrier with free delivery over 50 (tax 20%)
4) Create cart rule with free gift "Product 1" if the amount is more than 100 with 1 priority
5) Create cart rule with 40% discount with 2 priority

If I add "product 2" to cart, rule with free gift will be applied. After that I apply promocode for 40% discount. 
Thus, I get the total in the cart 130, a discount of 70, i.e. the total order amount is 60. But the shipping cost is calculated as paid.
It looks like after applying a discount promo code, it calculates the order amount without tax (that is, 48 if tax is 20%, not 60).
I tried a promotional code for exactly 40 euros, in this case the delivery is correctly calculated as free. 

What could be the reason for this behavior?
![Screenshot_5](https://github.com/PrestaShop/PrestaShop/assets/55474095/fb4f29a8-65bb-4a52-8fd5-e852a621c413)

**Comment by RosaBenouamer** [comment]
Hello @AnnRise
Thank you
Indeed it seems to be an issue !
I will convert the discussion as an issue :-)

**Comment by RosaBenouamer** [comment]
@AnnRise how do you parameter the minimum amount in condition tab ?
![image](https://github.com/PrestaShop/PrestaShop/assets/124665010/4801abd6-39b5-46a4-bcaf-66efec3f8b94)
Tax included or excluded

**Reply by RosaBenouamer** [reply]
@AnnRise could you please printscreen what do you parametered for point 3
Create carrier with free delivery over 50 (tax 20%)
For my first reading I understood you made a cart rules for this point ? Is it the case ?
Or is it in Shipping>Preference or Shipping>carrier ?

**Comment by RosaBenouamer** [comment]
I have created an issue https://github.com/PrestaShop/PrestaShop/issues/35135 so I close the discussion.
